store_flakes

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import ssl
+import sqlite3
+import urllib.request
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Store information about flaky tests')
+    parser.add_argument("--no-check-certificate", action="store_true", help="Don't validate the server's certificate")
+    parser.add_argument("--db", default="cockpit-flakes.db", help="Database name")
+    parser.add_argument('log', help="File log file")
+    opts = parser.parse_args()
+
+    db_conn = sqlite3.connect(opts.db)
+    cursor = db_conn.cursor()
+
+    init_db(cursor)
+
+    ctx = ssl.create_default_context()
+    if opts.no_check_certificate:
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
+
+    if opts.log.startswith("http"):
+        fp = urllib.request.urlopen(opts.log, context=ctx)
+    else:
+        fp = open(opts.log, "r")
+
+    for line in find_failed_tests(fp):
+        (test, name, retry) = process_line(line)
+        insert(cursor, test, name, 1 if retry else 0)
+
+    fp.close()
+
+    db_conn.commit()
+    db_conn.close()
+
+
+def init_db(cursor):
+    cursor.execute("CREATE TABLE if not exists CockpitFlakes (test text, name text, retry integer)")
```

Comment:
It's not clear what "test" and "name" mean. This is also not just for "Cockpit".

I think for meaningful queries we want to store some more information. E. g. we want something like "get me all the test logs where TestFoo.testBar() failed". So we should have one table `TestRuns` like
```sql
id INTEGER PRIMARY_KEY AUTOINCREMENT,
project CHAR[50],
context CHAR[50],
url VARCHAR(1000)
```

It could use our sink identifiers like `pull-895-20200526-085151-0f1159c6-fedora-31-aws-weldr-lorax` as ID, but just an auto-generated integer will be is faster for processing, and ultimately easier to use.

Then we want to record which tests failed in these runs, in a `Failures` table:
```sql
testname VARCHAR(100) NOT NULL,
retry_reason VARCHAR(1000),
run INTEGER NOT NULL,
FOREIGN KEY (run) REFERENCES TestRuns(id)
```
This also turns the retry reason from a boolean into a string which can be `NULL` (no retry), and we don't lose the information why it was retried (infrastructure or test failure)

WDYT?